File: examples/stack-samples/issue-145-php/app/Http/Controllers/UserController.php
<?php

namespace App\Http\Controllers;

use App\Services\UserService;

class UserController
{
    private UserService $userService;

    public function __construct(UserService $userService)
    {
        $this->userService = $userService;
    }

    public function index(): array
    {
        return $this->userService->getAllUsers();
    }

    public function show(int $id): ?array
    {
        $user = $this->userService->findUser($id);
        return $user ? ['name' => $user->getName()] : null;
    }

    public function store(string $name, string $email): array
    {
        $user = $this->userService->createUser($name, $email);
        return ['id' => $user->getId()];
    }

    public function destroy(int $id): bool
    {
        return $this->userService->deleteUser($id);
    }
}

File: examples/stack-samples/issue-144-java/src/main/java/com/example/userapi/controller/UserController.java
package com.example.userapi.controller;

import java.util.List;
import com.example.userapi.model.User;
import com.example.userapi.service.UserService;

public class UserController {
    private final UserService userService;

    public UserController(UserService userService) {
        this.userService = userService;
    }

    public List<User> listUsers() {
        return userService.getAllUsers();
    }

    public User getUser(Long id) {
        return userService.getUserById(id)
            .orElseThrow(() -> new RuntimeException("User not found"));
    }

    public User createUser(String name, String email) {
        return userService.createUser(name, email);
    }

    public void deleteUser(Long id) {
        userService.deleteUser(id);
    }
}
